scripts/performance/tracer.py

```diff
@@ -0,0 +1,93 @@
+from logging import getLogger
+
+class TracingStateManager:
+    '''A class to manage the state of tracing.'''
+    def __init__(self):
+        self.trace_provider_initialized = False
+        self.trace_provider_console_exporter_enabled = False
+        self.trace_opentelemetry_imported = False
+    
+    def set_initialized(self, value : bool): self.trace_provider_initialized = value
+    def set_console_exporter_enabled(self, value : bool): self.trace_provider_console_exporter_enabled = value
+    def set_opentelemetry_imported(self, value : bool): self.trace_opentelemetry_imported = value
+    def get_initialized(self): return self.trace_provider_initialized
+    def get_console_exporter_enabled(self): return self.trace_provider_console_exporter_enabled
+    def get_opentelemetry_imported(self): return self.trace_opentelemetry_imported
+
+tracing_state_manager = TracingStateManager()
+try:
+    from opentelemetry import trace
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+    tracing_state_manager.set_opentelemetry_imported(True)
+
+except ImportError:
+    pass
+
+def setup_tracing():
+    '''Set up the OpenTelemetry trace provider.'''
+    if tracing_state_manager.get_initialized() or not tracing_state_manager.get_opentelemetry_imported():
```

Comment:
We are checking here to see if the `tracing_state_manager` has been initialized, but from this PR, I don't see us ever setting that value. So this will always be false. Is that intended?